src/Microsoft.Azure.WebJobs.Host/Queues/Bindings/QueueBindingProvider.cs

```diff
@@ -80,19 +81,30 @@ public void Initialize(ExtensionConfigContext context)
 
                 binding.SetPostResolveHook(ToReadWriteParameterDescriptorForCollector)
                         .BindToInput<CloudQueue>(builder);
+            }
+
+            private async Task<IStorageQueueMessage> ConvertPocoToCloudQueueMessage(object arg, Attribute attrResolved, ValueBindingContext context)
+            {
+                var attr = (QueueAttribute)attrResolved;
```

Comment:
Since this hangs off of a binding with known TAttribute, can we make it pass TAttribute to the func? Then you wouldn't have to cast.

Reply:
I'll try - the problem I'm running into is "delegate overload hell".  We have too many permutations, and it stresses C#'s overload rules. 

---
In reply to: [148656706](https://github.com/Azure/azure-webjobs-sdk/pull/1413#discussion_r148656706) [](ancestors = 148656706)